transformations.go

```diff
@@ -0,0 +1,238 @@
+package main
+
+import (
+	"math/rand"
+	"strings"
+	"time"
+
+	weightedRand "github.com/mroth/weightedrand"
+)
+
+// Codon holds information for a codon triplet in a struct
+type Codon struct {
+	Triplet    string
+	Occurrence int
+}
+
+// AminoAcid holds information for an amino acid and related codons in a struct in a struct
+type AminoAcid struct {
+	Letter string
+	Codons []Codon
+}
+
+// CodonTable holds information for a codon table.
+type CodonTable struct {
+	StartCodons []string
+	StopCodons  []string
+	AminoAcids  []AminoAcid
+}
+
+// Translate translates a codon sequence to an amino acid sequence
+func Translate(sequence string, codonTable CodonTable) string {
+
+	var aminoAcids strings.Builder
+	var currentCodon strings.Builder
+	translationTable := codonTable.generateTranslationTable()
+
+	for _, letter := range sequence {
+
+		// add current nucleotide to currentCodon
+		currentCodon.WriteRune(letter)
+
+		// if current nucleotide is the third in a codon translate to aminoAcid write to aminoAcids and reset currentCodon.
+		if currentCodon.Len() == 3 {
+			aminoAcids.WriteString(translationTable[currentCodon.String()])
+
+			// reset codon string builder for next codon.
+			currentCodon.Reset()
+		}
+	}
+	return aminoAcids.String()
+}
+
+// Optimize takes an amino acid sequence and CodonTable and returns an optimized codon sequence
+func Optimize(aminoAcids string, annotatedSequence AnnotatedSequence, codonTable CodonTable) string {
+	var codons strings.Builder
+	var sequenceBuffer strings.Builder
+	for _, feature := range annotatedSequence.Features {
+		if feature.Type == "CDS" {
+			sequenceBuffer.WriteString(feature.getSequence())
+		}
+	}
+	optimizationTable := codonTable.generateOptimizationTable(sequenceBuffer.String())
+	for _, aminoAcid := range aminoAcids {
+		codons.WriteString(optimizationTable[string(aminoAcid)].Pick().(string))
+	}
+	return codons.String()
+}
+
```

Comment:
Optimization table should be an input to the `Optimize` function. This is because the AnnotatedSequences may be huge (in case of human chromosomes), and you will need to load multiple files in order to get the correct optimization table (again, with multiple human chromosomes). Once generated, you won't need to do that again.

Ideally, there would be some kind of robust import / export function to use JSON codon tables. But at minimum, split this function so that can be added later.